### PR TITLE
swatch-667 remove references to accountNumber in SubscriptionSyncController#findSubscriptionsAndSyncIfNeeded and callers

### PR DESCRIPTION
**Pull request by vbusch**
Jira issue: [SWATCH-667](https://issues.redhat.com/browse/SWATCH-667)

## Description
Remove references to accountNumber in SubscriptionSyncController#findSubscriptionsAndSyncIfNeeded and callers
* Removed subscription-api-spec.yaml:  SearchSubscriptionsByAccountNumber
* Removed the OracleAccountNumber
* Removed Subscription.accountNumber 

## Testing
Unit tests & Regression

### CODE REVIEW

File: swatch-producer-aws/src/main/java/com/redhat/swatch/aws/processors/BillableUsageProcessor.java [hunk 1]
```diff
@@ -88,9 +88,6 @@ public void process(BillableUsage billableUsage) {
     if (billableUsage.getOrgId() != null) {
       MDC.put("org_id", billableUsage.getOrgId());
     }
-    if (billableUsage.getAccountNumber() != null) {
```
Comment (Sgitario) [inline swatch-producer-aws/src/main/java/com/redhat/swatch/aws/processors/BillableUsageProcessor.java:91]: The property `account_number` should be removed from the generated class BillableUsage in: swatch-core/schemas/billable_usage.yaml.

Moreover, I've seen that swatch-core/schemas/event.yaml still contains the `account_number` property.

**Review by Sgitario — CHANGES_REQUESTED** [review]
There are a couple of places where the accountNumber still exists.

File: swatch-core/src/main/java/org/candlepin/subscriptions/db/model/Subscription.java [hunk 2]
```diff
@@ -85,9 +85,6 @@ public class Subscription implements Serializable {
   @Column(name = "billing_account_id")
   private String billingAccountId;
 
-  @Column(name = "account_number")
-  private String accountNumber;
```
[the remark this answers is not in the document]
Comment (vbusch) [reply swatch-core/src/main/java/org/candlepin/subscriptions/db/model/Subscription.java:89]: Yes, I created SWATCH-1811 for the migration script.  It should be done after the code changes are deployed to stage & production.  This handles the rolling deployment and also let us rollback if needed.

File: swatch-producer-aws/src/main/java/com/redhat/swatch/aws/processors/BillableUsageProcessor.java (hunk 1 above)
Comment (vbusch) [reply swatch-producer-aws/src/main/java/com/redhat/swatch/aws/processors/BillableUsageProcessor.java:91]: I have added another commit to cover the billable_usage and events.

**Review by Sgitario — APPROVED** [review]
lgtm!

**Pull request merged**